File: CS 471/Control Character Blocking/ControlChar.py
#Joseph Camacho-Terrazas
#10/26/2020
#Input: The backup text file
#Output: Resulting string after eliminating control characters
#Preconditions: The file must exist and filename must be valid
#Postconditions: Print the resulting string after eliminating all control characters

import string

#create a flag that will allows us to print
printFlag = 1
#Open the input and output files
f1 = open("control-char.txt")
f2 = open("pythonoutput.txt", "w")

while True:
    #Read one character of the file into c
    c = f1.read(1)
    
    #Break out of the loop at the end of the file
    if not c:
        break
    #Convert the chars to ascii values, and compare to control char value
    #If current char is ctrl-c, print flag is 0
    if ord(c) == 3:
        printFlag = 0
    #If current char is ctrl-b, print flag is 1 and skip to next iteration
    if ord(c) == 2:
        printFlag = 1
        continue
    #If print flag is 1, then write it to the output file
    if printFlag == 1:
        f2.write(c)

#User confirmation about the output file creation
print ("File created successfully ")
print (f2.name)

#Close all open files
f1.close()
f2.close()

File: CS 471/Control Character Blocking/ControlChar.java
/*Joseph Camacho-Terrazas
*10/26/2020
*Input: The backup text file
*Output: Resulting string after eliminating control characters
*Preconditions: The file must exist and filename must be valid
*Postconditions: Print the resulting string after eliminating all control characters
*/

import java.util.*;
import java.io.*;

public class ControlChar {
    public static void main(String[] args) {   
        try {
            //Declare the input and output files
            File input = new File("control-char.txt");
            File output = new File("javaoutput.txt");
            //Create a file reader, file writer, and buffered reader
            FileReader fr = new FileReader(input);
            FileWriter fw = new FileWriter(output);
            BufferedReader br = new BufferedReader(fr);
            //c stores value from buffered reader
            int c = 0;
            //Print flag allows us to print when it's 1
            int printFlag = 1;

            //Loop through the file using buffered reader
            while ((c = br.read()) != -1) {
                //Buffered reader gives us ints, so compare ascii values
                //If c is ctrl-c, printFlag is 0
                if(c == 3) {printFlag = 0;}
                //If c is ctrl-b, printFlag is 1 and skip to next iteration
                if(c == 2) {printFlag = 1; continue;}
                //If printFlag is 1, write c to the file
                if(printFlag == 1) {
                    fw.write(c);
                }
            }

            //close all open readers and writers
            fr.close();
            fw.close();
            br.close();
        }
    
        catch (IOException e) {
            //Catch any IO Exceptions
            System.out.println("There was an error" + e);
        }
    }
}